### PR TITLE
Feature/app repository

### PR DESCRIPTION
**Pull request by baron1xxx**

### CODE REVIEW

File: backend/src/services/applicationService.ts [hunk 1]
```diff
@@ -1,40 +1,70 @@
...
+  }
+};
 
-export const addApp = ({ appName }: any) => {
-  const appId = new Date().getTime().toString();
-  const name: string = appName;
-  const app = { id: appId, name, organizationId: `o${appId}`, updatedByUserId: `u${appId}` };
-  apps.push(app);
-  return Promise.resolve(app);
+export const addApp = async ({ name, organizationId, updatedByUserId }: any) => {
```
Comment (uliana-lobanova) [inline backend/src/services/applicationService.ts]: get rid of any

File: backend/src/services/applicationService.ts [hunk 2]
```diff
@@ -1,40 +1,70 @@
...
+  try {
+    const app = await getCustomRepository(ApplicationRepository).getAppByName(name);
```
Comment (uliana-lobanova) [inline backend/src/services/applicationService.ts]: We need to check not only by name but also by user's currentOrganizationId. Two different organizations can have apps with the same name. You can get user's info from `req.user`

File: backend/src/data/repositories/applicationRepository.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { IEditApplication } from '../../common/models/application/IEditApplication';
+
+@EntityRepository(App)
+export class ApplicationRepository extends Repository<App> {
+  addApp(application: ICreateApplication): Promise<App> {
+    return this.create(application).save();
+  }
+
+  getAllApp(): Promise<Array<App>> {
+    return this.find({ relations: ['organization', 'updatedByUser'] });
```
Comment (uliana-lobanova) [inline backend/src/data/repositories/applicationRepository.ts]: you need to get all apps of user's current organization, not just all apps

**Review by uliana-lobanova — CHANGES_REQUESTED** [review]
Also:
1) Set the return type for all your service functions and make sure you return the data exactly of that type. Currently, you return a lot of redundant columns to the client
2) No need to wrap your service functions into `try...catch` expression. All errors should be caught in the errorHandlerMiddleware

**Review by AlikRakhmonov — APPROVED** [review]

**Review by uliana-lobanova — APPROVED** [review]

**Pull request merged**